src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -128,6 +133,112 @@ internal partial class Binder
+            BoundObjectInitializerExpressionBase boundInitializerOpt = node.Initializer != null
+                ? BindInitializerExpression(syntax: node.Initializer, type: strippedType, typeSyntax: syntax, diagnostics)
+                : null;
+
+            BoundExpression operand = source;
+            bool hasErrors;
+            switch (strippedType.TypeKind)
+            {
+                case TypeKind.Array:
+                case TypeKind.Enum:
+                case TypeKind.Delegate:
+                case TypeKind.Interface:
+                    Error(diagnostics, ErrorCode.ERR_BadTargetTypeForNew, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Pointer:
+                    Error(diagnostics, ErrorCode.ERR_UnsafeTypeInObjectCreation, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Dynamic:
+                    Error(diagnostics, ErrorCode.ERR_NoConstructors, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Struct when strippedType.IsTupleType:
+                    Error(diagnostics, ErrorCode.ERR_NewWithTupleTypeSyntax, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Struct:
+                case TypeKind.Class:
+                    operand = BindClassCreationExpression(
+                        node,
+                        typeName: strippedType.Name,
+                        typeNode: node,
+                        type: (NamedTypeSymbol)strippedType,
+                        arguments,
+                        diagnostics,
+                        boundInitializerOpt,
+                        forTargetTypedNew: true);
+                    operand.WasCompilerGenerated = true;
+                    hasErrors = operand.HasErrors;
+                    break;
+
+                case TypeKind.TypeParameter:
+                    var typeParameter = (TypeParameterSymbol)strippedType;
+                    if (!typeParameter.IsInstantiable())
+                    {
+                        Error(diagnostics, ErrorCode.ERR_BadTargetTypeForNew, syntax, strippedType);
+                        hasErrors = true;
+                        break;
+                    }
+
+                    operand = BindTypeParameterCreationExpression(
+                        node,
+                        typeParameter,
+                        boundInitializerOpt,
+                        diagnostics);
+                    operand.WasCompilerGenerated = true;
+                    hasErrors = operand.HasErrors;
+                    break;
+
+                default:
+                    hasErrors = true;
+                    break;
+            }
+
+            arguments.Free();
+
+            var result = new BoundConversion(
+                syntax,
```

Comment:
I don't think this node belongs in the resulting bound tree. We don't have, for example, a bound node for the literal conversion from "3" to `byte`. Such conversions are (one might argue prematurely) lowered to their meaning in initial binding. The conversion is a trick of the language definition, not something that the programmer intends to occur.